pkg/fs/fs.go

```diff
@@ -25,3 +27,38 @@ type DiskStatus struct {
 	Free  uint64
 	Avail uint64
 }
+
+// MoveFileWithReplacement copies the file contents at `src` to `dst`.
+//
+// If the file at `dst` already exists, it will be truncated and its contents
+// overwritten.
+func MoveFileWithReplacement(src, dst string) error {
+	out, err := os.Create(dst)
+	if err != nil {
+		return err
+	}
+
+	in, err := os.Open(src)
+	if err != nil {
+		out.Close()
+		return err
+	}
+
+	defer in.Close()
```

Comment:
It would be good to capture and return any error from this `in.Close()` call. You can name the function's error return and use the `Capture` function here to ensure the error is handled: https://github.com/influxdata/influxdb/blob/master/pkg/errors/error_capture.go

I would also recommend that you make all the `out.Close` calls in a deferred `Capture` as well, from a nested local function that completes before the `os.Remove` call.  This will eliminate redundant code.

Reply:
I added a ``errors.Capture()`` now for the ``in.Close()`` call. I'm not 100% sure this is the right approach though. Background is that we don't really care if ``in.Close()`` succeeds, since we can't do anything about it and because it's a file descriptor only opened for reading. So if the close fails for some reason, then the operation was successful, but still the (unlikely) error from close would be returned and the operation would count as failed. Not sure if that's what you guys want so please double check.

I also moved the copy part to its own function, so that the defer is more "natural" there and does not require special checks to see that it happens before the remove.

Reply:
My general policy is to check all return codes, because a failure is often the first indicator of something larger going awry.  The failure of `Close` here may not affect the current operation, but it certainly may be an indication that your system is unhappy in a way that may be or may become serious.

Reply:
Alright, then the diff should be fine like that.